source/tryGoal/component/tryGoalController.js

```diff
@@ -112,6 +113,28 @@ function($scope, agentMemory, $compile, $stateParams, config, GoalAPI, ConfigAPI
         $scope.$parent.splitScreen = !$scope.$parent.splitScreen;
     };
 
+    $scope.toggleQuestionContext = function () {
+        $scope.$parent.displayQuestionContext = !$scope.$parent.displayQuestionContext;
+    }
+
+    function formatConcept(concept) {
+        if (concept === '(null)') {
+            return '(*)';
+        } else {
+            return concept;
+        }
+    };
+
+    $scope.formatRule = function (rule) {
+        var concepts = rule.match(/\((.*?)\)/gm);
+        var relationship = rule.match(/\[(.*?)\]/);
+
+        concepts[0] = formatConcept(concepts[0]);
+        concepts[1] = formatConcept(concepts[1]);
```

Comment:
Can we be certain that these will always exist?

Reply:
There aren't any checks for response.ruleStack specifically, but we do check whether response.question exists in order to change the display. 
I think it would be super unlikely to not have the rule stack there after checking that the question exists, but it's probably worth having a contingency. I'll wrap it in something :)